### ISSUE TITLE
Stop Using Reflection in the LSP Request Invoker

### ISSUE THREAD

**Issue by TanayParikh** [dotnet/aspnetcore#23191]
With https://github.com/dotnet/aspnetcore-tooling/pull/2081 we've brought in `Microsoft.VisualStudio.LanguageServer.Client.Implementation`, and consequently we shouldn't have to rely on reflection to get the appropriate request methods from the service broker.

**Comment by TanayParikh** [comment]
Not going to be possible for now, because the [`SynchronizedRequestAsync`](https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/DefaultLSPRequestInvoker.cs#L49-L58) method we use is not part of the `ILanguageServiceBroker2` interface which we gained access through with `Microsoft.VisualStudio.LanguageServer.Client.Implementation`.

**Comment by TanayParikh** [comment]
Unblocked by: https://github.com/dotnet/aspnetcore-tooling/pull/2267
